### PR TITLE
Refactor stringify, and snapshot output

### LINKED ISSUE #216: Snapshot stringify output and run code coverage on stringify

### LINKED ISSUE #217: Expose HostValue.protoype.stringify(ctx)
Export context class from core too.

### PR DESCRIPTION
**Pull request by jtenner**
fixes #216 
fixes #217 

- [x] make options optional
- [x] refactor stringify method
- [x] add snapshots to jest tests for stringify output
- [x] get code coverage on stringifyHostValue.ts
- [x] increase code coverage to 100%

### CODE REVIEW

**Comment by jtenner** [comment]
@willemneal we can start snapshotting the output for the stringify functions. Can you help me look through `packages/core/assembly/jest-log.ts` and add some tests for output?

Ideas:
1. Really deeply nested object (should stop at level 5 and I don't think it does)
2. Arrays with more than `maxItemCount`
3. Objects with keys greater than `maxItemCount`
etc

File: packages/core/__tests__/TestContext.log.spec.ts [hunk 1]
```diff
@@ -17,6 +18,15 @@ let start = new Promise<void>((resolve, reject) => {
 
 beforeEach(() => start);
 
+const stringifyOptions: Partial<StringifyHostValueProps> = {
+  indent: 2,
+  tab: 4,
+  classNameColor: (name) => "class: " + name,
```
Comment (DuncanUszkay1) [inline packages/core/__tests__/TestContext.log.spec.ts]: In what sense is this a color?

Comment (jtenner) [reply packages/core/__tests__/TestContext.log.spec.ts]: For snapshot output. In gonna change that soon.

File: packages/core/__tests__/TestContext.log.spec.ts [hunk 2]
```diff
@@ -17,6 +18,15 @@ let start = new Promise<void>((resolve, reject) => {
...
+  keywordColor: (keyword) => "keyword: " + keyword,
+  numberColor: (number) => "number: " + number,
```
Comment (DuncanUszkay1) [inline packages/core/__tests__/TestContext.log.spec.ts]: would `numberFormatter` make more sense here? This line makes it pretty clear that this property can do more than change the color

Comment (jtenner) [reply packages/core/__tests__/TestContext.log.spec.ts]: Absolutely. That sounds fine to me.

**Comment by jtenner** [comment]
```
stringifyHostValue.ts |   78.03 |    59.09 |   52.17 |   80.49 | 73,79,85,94,125,166,175,212,264,267,277,302,341,355,380,388-392,395,398,400,402,405-411,430 
```

😅

**Pull request merged**